dataflux_pytorch/benchmark/checkpointing/simulated/README.md

```diff
@@ -0,0 +1,115 @@
+
+### Configuration
+
+Ensure you are running within a python virtual environment, and ensure gcloud config project is set to your project. You can use the following command to set it:
+
+```shell
+gcloud config set project {PROJECT_ID}
+```
+
+### Environment variables:
+
+Set the following environment variables by updating the deployment file to run on a GKE cluster.
+1. Set the environment variables required to run the demo. These include:
+  
+  * `PROJECT`: The GCP project you're using.
+  
+  * `CKPT_DIR_PATH`: The full path of the directory in which to save checkpoints, in the format `gs://<bucket>/<directory>/`
+
+  * `WORLD_SIZE`: The number of nodes.
+
+  * `LAYER_SIZE`: The size of each layer.
+
+2. Set the optional environment variables, if desired:
+  
+  * `PADDING_SIZE`: The number of dummy tensors to add to the state_dict in order to produce checkpoints of desired size.
+    *   Both padding_size and layer_size will impact the size of the checkpoint. Therefore, set layer_size to an appropriate value first, and then adjust padding_size until the checkpoint of the desired size is generated. The default value for padding_size is 4000.
+  
+  * `SAMPLE_COUNT`: The number of times save_checkpoint/load_checkpoint is called. The default is 8.
+
+  * `USE_FSSPEC`: If set to true, the code will use [gcsfs/fsspec](https://github.com/fsspec/gcsfs) in order to save_checkpoint/restore_checkpoint from GCS.
+    *   If not set, it will use GCS Connector for Pytorch by default.
+
+### Installing Requirements:
+ 
+* Install the required dependencies using the following commands:
+
+```shell
+ pip install -r dataflux_pytorch/benchmark/requirements.txt; 
+ pip install .
+```
+### GCS Auth
+
+Run the following commands to authenticate your application so it can access GCS buckets.
+```shell
+gcloud config set project PROJECT_ID
+gcloud auth application-default login
+```
+
+### Running
+
+_Note: The following instructions assume that you have Jobset and Kueue enabled on your GKE cluster. For easier compatibility, we recommend creating a cluster using [XPK](https://github.com/google/xpk) which will configure these features automatically._
+
+1.  To connect to GKE, you need to generate a kubeconfig file. This can be done using the following command:
+
+```shell
+gcloud container clusters get-credentials {YOUR-GKE-CLUSTER-NAME} --zone {ZONE} --project {YOUR-GCP-PROJECT}
+```
+
+2.  Build the container.
+
+Ensure that your working directory is `dataflux-pytorch`
+```shell
+docker build -t {YOUR_CONTAINER_NAME} .
+```
+
+3.  Upload your container to the container registry:
+
+```shell
+docker tag {YOUR_CONTAINER_NAME} gcr.io/{YOUR-GCP-PROJECT}/{YOUR_CONTAINER_NAME}
+docker push gcr.io/{YOUR-GCP-PROJECT}/{YOUR_CONTAINER_NAME}
+```
+
+4.  Update the values in `benchmark-deployment.yaml`.
+
+5.  Deploy the configuration using the following command:
+
+```shell
+kubectl apply -f /app/dataflux_pytorch/benchmark/checkpointing/simulated/benchmark-deploy.yaml
```

Comment:
If the user's working directory is `dataflux-pytorch` I don't think this path will have a leading `/app/`

Reply:
The apply command typically isn't run by docker, but instead run locally by the user to apply a configuration to a cluster. I don't think most users would have the /app directory.

Reply:
I just ran the the GKE demo using deployment-yaml file right now. Seems like it is working as expected (and the path is correct).

Reply:
Typically the apply command is run from the users personal machine, is it not? What did your execution command look like? Usually the `/app` directory is just built into the image itself and is used for code that's running within the docker container.

Reply:
Oh I was completely misreading the readme. Corrected. 

Thanks for catching this @MattIrv